src/main/scala/cromwell/engine/backend/Backend.scala

```diff
@@ -0,0 +1,28 @@
+package cromwell.engine.backend
+
+import cromwell.binding.WdlExpression.LookupFunction
+import cromwell.binding.values.WdlValue
+import cromwell.binding.{Call, TaskOutput}
+import cromwell.engine.SymbolStore
+
+import scala.util.Try
+
+/**
+ * trait to be implemented by concrete backends
+ */
+trait Backend {
+
+  /** Curried version that a caller is expected to fix for a Call and symbol store. */
+  private def curriedLookupFunction(call: Call, symbolStore: SymbolStore)(localName: String): WdlValue =
```

Comment:
Like the 'maybe' business, I'd prefer not saying "curried". the signature shows it's curried.